server/base/src/main/java/org/apache/accumulo/server/master/balancer/HostRegexTableLoadBalancer.java

```diff
@@ -101,22 +105,86 @@
+            continue;
+          }
+          String tableName = customProp.getKey().substring(HOST_BALANCER_PREFIX.length());
+          String regex = customProp.getValue();
+          regexes.put(tableName, regex);
+        }
+      }
+    }
+
+    return ImmutableMap.copyOf(regexes);
+  }
+
+  /**
+   * Host Regex Table Load Balance Config
+   */
+  static class HrtlbConf {
+
+    protected long oobCheckMillis =
+        ConfigurationTypeHelper.getTimeInMillis(HOST_BALANCER_OOB_DEFAULT);
+    private int maxTServerMigrations = HOST_BALANCER_REGEX_MAX_MIGRATIONS_DEFAULT;
+    private int maxOutstandingMigrations = DEFAULT_OUTSTANDING_MIGRATIONS;
+    private boolean isIpBasedRegex = false;
+    private Map<String,String> regexes;
+    private Map<String,Pattern> poolNameToRegexPattern = null;
+
+    HrtlbConf(AccumuloConfiguration aconf) {
+      System.out.println("building hrtlb conf");
+      String oobProperty = aconf.get(HOST_BALANCER_OOB_CHECK_KEY);
+      if (oobProperty != null) {
+        oobCheckMillis = ConfigurationTypeHelper.getTimeInMillis(oobProperty);
+      }
+      String ipBased = aconf.get(HOST_BALANCER_REGEX_USING_IPS_KEY);
+      if (ipBased != null) {
+        isIpBasedRegex = Boolean.parseBoolean(ipBased);
+      }
+      String migrations = aconf.get(HOST_BALANCER_REGEX_MAX_MIGRATIONS_KEY);
+      if (migrations != null) {
+        maxTServerMigrations = Integer.parseInt(migrations);
+      }
+      String outstanding = aconf.get(HOST_BALANCER_OUTSTANDING_MIGRATIONS_KEY);
+      if (outstanding != null) {
+        maxOutstandingMigrations = Integer.parseInt(outstanding);
+      }
+
+      this.regexes = getRegexes(aconf);
+
+      Map<String,Pattern> poolNameToRegexPatternBuilder = new HashMap<>();
+      regexes.forEach((k, v) -> {
+        poolNameToRegexPatternBuilder.put(k, Pattern.compile(v));
+      });
+
+      poolNameToRegexPattern = ImmutableMap.copyOf(poolNameToRegexPatternBuilder);
+    }
+  }
 
   private static final long ONE_HOUR = 60 * 60 * 1000;
   private static final Set<KeyExtent> EMPTY_MIGRATIONS = Collections.emptySet();
-
-  private volatile Map<TableId,String> tableIdToTableName = null;
-  private volatile Map<String,Pattern> poolNameToRegexPattern = null;
+  // private volatile Map<TableId,String> tableIdToTableName = null;
+  // private volatile Map<String,Pattern> poolNameToRegexPattern = null;
   private volatile long lastOOBCheck = System.currentTimeMillis();
-  private volatile boolean isIpBasedRegex = false;
   private Map<String,SortedMap<TServerInstance,TabletServerStatus>> pools = new HashMap<>();
-  private volatile int maxTServerMigrations = HOST_BALANCER_REGEX_MAX_MIGRATIONS_DEFAULT;
-  private volatile int maxOutstandingMigrations = DEFAULT_OUTSTANDING_MIGRATIONS;
   private final Map<KeyExtent,TabletMigration> migrationsFromLastPass = new HashMap<>();
   private final Map<String,Long> tableToTimeSinceNoMigrations = new HashMap<>();
 
+  // TODO volatile?
```

Comment:
TODO should be resolved one way or the other, rather than left in the code. It can always be adjusted later if the decision to make it volatile or not is reversed.